### PR TITLE
Support for MKL82Z

### LINKED ISSUE #300: Support for NXP FRDM-KL82Z
I'm currently working on a project that uses rosserial to communicate with a NXP FRDM-KL82Z, and I was wondering if I should create a pull request to this repository to add support for this board. 

Is this something that you are interested in?

### PR DESCRIPTION
**Pull request by sindrehan**
This PR adds rosserial support for the MKL82Z chips from NXP. 
The UART driver depends on the SDK from NXP, I have added a note about that in the readme. 
Please review and let me know if there are any issues I need to fix, or anything else that needs to be done before merging.

Closes #300

### CODE REVIEW

**Comment by sindrehan** [comment]
Not sure why CI is failing, seems to be some error with rosdep: 
```
$ rosdep install --from-paths src --ignore-src --rosdistro=$CI_ROS_DISTRO -y
dpkg-query: no packages found matching ros-indigo-rospy
dpkg-query: no packages found matching ros-indigo-roscpp
dpkg-query: no packages found matching ros-indigo-geometry-msgs
dpkg-query: no packages found matching ros-indigo-std-msgs
dpkg-query: no packages found matching libgtest-dev
dpkg-query: no packages found matching ros-indigo-message-runtime
dpkg-query: no packages found matching ros-indigo-rostest
dpkg-query: no packages found matching ros-indigo-catkin
dpkg-query: no packages found matching ros-indigo-message-generation
dpkg-query: no packages found matching ros-indigo-diagnostic-msgs
dpkg-query: no packages found matching ros-indigo-sensor-msgs
dpkg-query: no packages found matching arduino-core
dpkg-query: no packages found matching ros-indigo-topic-tools
dpkg-query: no packages found matching ros-indigo-tf
dpkg-query: no packages found matching ros-indigo-rosbash
dpkg-query: no packages found matching ros-indigo-nav-msgs
ERROR: the following packages/stacks could not have their rosdep keys resolved
to system dependencies:
rosserial: Missing resource rosserial
```

**Comment by sindrehan** [comment]
@mikepurvis Any feedback on this?

File: rosserial_mkl82z/package.xml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  <description>Library for ROSserial usage on NXP MKL82Z</description>
+
+  <author>Sindre Hansen</author>
+  <maintainer email="sindrehansen92@gmail.com">Sindre Hansen</maintainer>
+
+  <license>BSD</license>
+
+  <url type="website">http://wiki.ros.org/rosserial_mkl82z</url>
+
+  <buildtool_depend>catkin</buildtool_depend>
```
Comment (mikepurvis) [inline rosserial_mkl82z/package.xml:14]: Please add the appropriate dependencies here, see for example:

https://github.com/ros-drivers/rosserial/blob/jade-devel/rosserial_arduino/package.xml#L16-L22

**Comment by mikepurvis** [comment]
I'm not familiar with this hardware, but thanks for the contribution! Please fix the dependencies and then we can get this in.

**Comment by sindrehan** [comment]
We no longer use rosserial, so I'll go ahead and close this PR.

**Pull request closed without merging**